mayastor/src/bdev/nexus/nexus_share.rs

```diff
@@ -52,49 +31,43 @@ impl Share for Nexus {
                 .bdev
                 .share_iscsi()
                 .await
-                .map_err(|e| Error::ShareIscsiNexus {
-                    source: NexusIscsiError::CreateTargetFailed {
-                        dev: self.bdev.to_string(),
-                        err: e.to_string(),
-                    },
+                .map_err(|error| Error::ShareIscsiNexus {
+                    source: error,
                     name: self.name.clone(),
                 })
-                .map(|_u| self.share_uri().unwrap()),
-            Some(p) => {
-                error!("nexus {} already shared as {:?}", self.name, p);
-                Err(AlreadyShared {
+                .map(|_| self.share_uri().unwrap()),
+            Some(protocol) => {
+                error!("nexus {} already shared as {:?}", self.name, protocol);
+                Err(Error::AlreadyShared {
                     name: self.name.clone(),
                 })
             }
         }
     }
 
     async fn share_nvmf(&self) -> Result<Self::Output, Self::Error> {
-        let bdev = Bdev::from(self.bdev.as_ptr());
         match self.shared() {
             Some(Protocol::Nvmf) => Ok(self.share_uri().unwrap()),
-            Some(Protocol::Off) | None => bdev
+            Some(Protocol::Off) | None => self
+                .bdev
                 .share_nvmf()
                 .await
-                .map_err(|e| Error::ShareNvmfNexus {
-                    source: NexusNvmfError::CreateTargetFailed {
-                        dev: self.bdev.to_string(),
-                        err: e.to_string(),
-                    },
+                .map_err(|error| Error::ShareNvmfNexus {
+                    source: error,
                     name: self.name.clone(),
                 })
                 .map(|_| self.share_uri().unwrap()),
-            Some(p) => {
-                warn!("nexus {} already shared as {}", self.name, p);
-                Err(AlreadyShared {
+            Some(protocol) => {
+                warn!("nexus {} already shared as {}", self.name, protocol);
+                Err(Error::AlreadyShared {
                     name: self.name.clone(),
                 })
             }
         }
     }
 
     async fn unshare(&self) -> Result<Self::Output, Self::Error> {
-        self.bdev.unshare().await.map_err(|_e| Error::NotShared {
+        self.bdev.unshare().await.map_err(|_| Error::NotShared {
```

Comment:
I know it was already there, but that `Error::NotShared` does not seem correct?

Reply:
Yeah, have created a new Snafu error type for this